packages/animation/src/types.js

```diff
@@ -0,0 +1,85 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { FlyInEffectInputPropTypes } from './effects/flyIn/animationProps';
+import { PanEffectInputPropTypes } from './effects/pan/animationProps';
+import { PulseEffectInputPropTypes } from './effects/pulse/animationProps';
+import { RotateInEffectInputPropTypes } from './effects/rotateIn/animationProps';
+import { WhooshInEffectInputPropTypes } from './effects/whooshIn/animationProps';
+import { ZoomEffectInputPropTypes } from './effects/zoom/animationProps';
+import { BasicAnimationInputPropTypes } from './parts/defaultAnimationProps';
+
+export const AnimationFormPropTypes = PropTypes.shape({
+  ...FlyInEffectInputPropTypes,
+  ...PanEffectInputPropTypes,
+  ...PulseEffectInputPropTypes,
+  ...RotateInEffectInputPropTypes,
+  ...WhooshInEffectInputPropTypes,
+  ...ZoomEffectInputPropTypes,
+  ...BasicAnimationInputPropTypes,
+});
+
+// TODO, load this from original source, once made into a package.
+export const StoryElementPropType = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  x: PropTypes.number.isRequired,
+  y: PropTypes.number.isRequired,
+  width: PropTypes.number.isRequired,
+  height: PropTypes.number.isRequired,
+  flip: PropTypes.shape({
+    vertical: PropTypes.bool,
+    horizontal: PropTypes.bool,
+  }),
+  rotationAngle: PropTypes.number.isRequired,
+  mask: PropTypes.shape({
+    type: PropTypes.string.isRequired,
+  }),
+  link: PropTypes.shape({
+    url: PropTypes.string.isRequired,
+    desc: PropTypes.string,
+    icon: PropTypes.string,
+  }),
+  opacity: PropTypes.number,
+  lockAspectRatio: PropTypes.bool,
+});
+
+/**
+ * Element object
+ *
+ * @typedef {Element} Element A story element
+ * @property {string} id  A unique uuid for the element
+ * @property {string} type The type of the element, e.g. video, gif, image
+ * @property {number} x The x position of the element, its top left corner
+ * @property {number} y The y position of the element, its top left corner
+ * @property {number} width The width of the element
+ * @property {number} height The height of the element
+ * @property {Object} flip If the element has been flipped vertical/horizontal
+ * @property {number} rotationAngle The element's rotation angle
+ * @property {Object} mask The type of mask applied to the element
+ * @property {Object} link The url, icon and description of a link applied to element
+ * @property {number} opacity The opacity of the element
+ * @property {boolean} lockAspectRatio Whether the element's aspect ratio is locked
+ * @property {Resource} resource The element's resource object
```

Comment:
Looks like the `Resource` type def needs to be copied as well. Or just replace with `Object` here, should be OK.

```suggestion
 * @property {Object} resource The element's resource object
```